src/components/chat/GithubRepoModal.tsx

```diff
@@ -0,0 +1,346 @@
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogDescription,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "@/components/ui/accordion";
+import { useEffect, useState } from "react";
+import { Loader2, Info, Github } from "lucide-react";
+import { useNavigate } from "@tanstack/react-router";
+import { IpcClient } from "@/ipc/ipc_client";
+import { Alert, AlertDescription } from "@/components/ui/alert";
+import { useStreamChat } from "@/hooks/useStreamChat";
+import { selectedAppIdAtom } from "@/atoms/appAtoms";
+import { useSetAtom } from "jotai";
+import { useSettings } from "@/hooks/useSettings";
+import { UnconnectedGitHubConnector } from "@/components/GitHubConnector";
+import { AI_RULES_PROMPT } from "@/components/ImportAppDialog";
+
+interface GithubRepoModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function GithubRepoModal({ isOpen, onClose }: GithubRepoModalProps) {
+  const [repos, setRepos] = useState<any[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [url, setUrl] = useState("");
+  const [importing, setImporting] = useState(false);
+  const [installCommand, setInstallCommand] = useState("pnpm install");
+  const [startCommand, setStartCommand] = useState("pnpm dev");
+  const navigate = useNavigate();
+  const { streamMessage } = useStreamChat({ hasChatId: false });
+  const setSelectedAppId = useSetAtom(selectedAppIdAtom);
+  const { settings, refreshSettings } = useSettings();
+
+  const isAuthenticated = !!settings?.githubAccessToken;
+
+  useEffect(() => {
+    if (isOpen) {
+      if (isAuthenticated) {
+        fetchRepos();
+      }
+      // Reset state when modal opens
+      setUrl("");
+      setError(null);
+      setInstallCommand("pnpm install");
+      setStartCommand("pnpm dev");
+    }
+  }, [isOpen, isAuthenticated]);
+
+  const fetchRepos = async () => {
+    setLoading(true);
+    setError(null);
+    try {
+      const fetchedRepos = await IpcClient.getInstance().listGithubRepos();
+      setRepos(fetchedRepos);
+    } catch (err: any) {
+      setError(err.message || "Failed to fetch repositories.");
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const checkAndGenerateAiRules = async (appId: number, appPath: string) => {
```

Comment:
i'm noticing even when i import an app **with** AI_RULES.md it's still trying to generate this file even though it already exists. could you take a look?